src/dune_rules/lock_rules.ml

```diff
@@ -0,0 +1,125 @@
+open Import
+open Memo.O
+
+module Spec = struct
+  type ('path, 'target) t =
+    { target : 'target
+    ; lock_dir : string
+    }
+
+  let name = "lock"
+  let version = 1
+  let bimap t _ g = { t with target = g t.target }
+  let is_useful_to ~memoize = memoize
+
+  let encode { target; lock_dir } _encode_path encode_target : Sexp.t =
+    Sexp.List [ encode_target target; Sexp.Atom lock_dir ]
+  ;;
+
+  let action { target; lock_dir } ~ectx:_ ~eenv:_ =
+    let open Fiber.O in
+    let+ () = Fiber.return () in
+    Printf.eprintf
+      "Running ACTION, target is %s, our lock_dir is %S\n"
+      (Path.Build.to_string target)
+      lock_dir;
+    let path = Path.build target in
+    Path.mkdir_p path;
+    let t = Unix.localtime @@ Unix.gettimeofday () in
```

Comment:
I don't think you can include this information here. It makes the action non deterministic unfortunately. You could make it part of the action input if you really want it, but i'd say it's best to skip it.

Reply:
I know. This is just for debugging, as I can't make the action rerun on changes in `dune-project` and want to have a way to see whether the rule was run or not. The final PR will not have this anymore.